inference-engine/include/ie_layouts.h

```diff
@@ -318,4 +318,36 @@ class INFERENCE_ENGINE_API_CLASS(TensorDesc) {
     BlockingDesc blockingDesc;
 };
 
+/**
+ * @brief This structure describes ROI data for image-like tensors.
+ */
+struct ROI {
+    size_t id = 0;      //!< ID of a ROI (offset over batch dimension)
+    size_t posX = 0;    //!< W upper left coordinate of ROI
+    size_t posY = 0;    //!< H upper left coordinate of ROI
+    size_t sizeX = 0;   //!< W size of ROI
+    size_t sizeY = 0;   //!< H size of ROI
+
+    ROI() = default;
+
+    ROI(size_t id, size_t posX, size_t posY, size_t sizeX, size_t sizeY) :
+        id(id), posX(posX), posY(posY), sizeX(sizeX), sizeY(sizeY) {
+    }
+};
+
+/**
+ * @brief Creates a TensorDesc object for ROI.
+ *
+ * @param origDesc original TensorDesc object.
+ * @param roi An image ROI object inside of the original object.
+ * @param useOrigMemDesc Flag to use original memory description (strides/offset).
+ *     Should be set if the new TensorDesc describes shared memory.
```

Comment:
what do you mean `describes shared memory`?

Reply:
as far as I can see, it's used only with `true` argument. Does it mean we can remove 3rd argument at least for now?

Reply:
It is for `RemoteBlob`. Since `TensorDesc` in it describes mapped host memory, it might not use the same strides as parent object, it might be compact. Maybe the wording is bad.